From 1156ace4091cbc504e535bda1dc2969aae088e97 Mon Sep 17 00:00:00 2001
From: David Li <li.davidm96@gmail.com>
Date: Fri, 31 Mar 2023 11:57:35 -0400
Subject: [PATCH 4/4] GH-34743: [Python] Relax condition in flaky Flight test
 (#34747)

### Rationale for this change
This test is consistently flaky on AMD64 macOS.

### What changes are included in this PR?

Relax the test condition a bit.
### Are these changes tested?

This is a change to a test.

### Are there any user-facing changes?

No.
* Closes: #34743

Authored-by: David Li <li.davidm96@gmail.com>
Signed-off-by: Will Jones <willjones127@gmail.com>
---
 python/pyarrow/tests/test_flight.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/pyarrow/tests/test_flight.py b/python/pyarrow/tests/test_flight.py
index 69318a553..a42681678 100644
--- a/python/pyarrow/tests/test_flight.py
+++ b/python/pyarrow/tests/test_flight.py
@@ -1958,7 +1958,7 @@ def test_generic_options():
         client = flight.connect(('localhost', s.port),
                                 tls_root_certs=certs["root_cert"],
                                 generic_options=options)
-        with pytest.raises(pa.ArrowInvalid):
+        with pytest.raises((pa.ArrowInvalid, flight.FlightCancelledError)):
             client.do_get(flight.Ticket(b'ints'))
         client.close()
 
-- 
2.40.1.windows.1

